### PR TITLE
Fix backend edge cases (#275)

### LINKED ISSUE #275: Handle host uninstalling Zoom app
If a host uninstalls the Zoom app, the only way I can figure out for them to reinstall it is:
1. Someone with superuser privileges goes in to the user's account in admin and replaces Backend Metadata with:
`{
    "zoom": {
    }
}` and saves
2. Host navigates to a queue that has Zoom as an allowed meeting type and re-authorizes

That's the only thing I can find that triggers them to re-authorize.

If they click Create Meeting on a Zoom meeting after they uninstall it, nothing happens and the generic Zoom error shows:
> Start Meeting: An unexpected error occurred in Zoom. You can check the ITS Status page (https://status.its.umich.edu/) to see if there is a known issue with Zoom, or contact the ITS Service Center (https://its.umich.edu/help) for help.

I'm not sure the correct fix for this...

### PR DESCRIPTION
**Pull request by jlost**
Closes #275

### CODE REVIEW

File: src/officehours_api/backends/zoom.py [hunk 1]
```diff
@@ -167,9 +174,7 @@ def save_user_meeting(cls, backend_metadata: dict, assignee: User):
         if backend_metadata.get('meeting_id'):
             return backend_metadata
 
-        user_email = backend_metadata['user_email']
-        user = User.objects.get(email=user_email)
-        meeting = cls._create_meeting(user)
+        meeting = cls._create_meeting(assignee)
```
Comment (jlost) [inline src/officehours_api/backends/zoom.py:177]: It looks like this was causing the meeting to always be created for the first host, which would often not be the assignee, but in testing scenarios, often *would* be.

File: src/officehours_api/models.py [hunk 2]
```diff
@@ -159,8 +159,6 @@ def start(self, assignee=None):
         if not self.assignee:
             raise Exception("Can't start meeting before assignee is set!")
         backend = BACKEND_INSTANCES[self.backend_type]
-        user_email = self.queue.hosts.first().email
-        self.backend_metadata['user_email'] = user_email
```
Comment (jlost) [inline src/officehours_api/models.py:163]: No longer needed, not used anywhere.

File: src/officehours_api/backends/zoom.py (hunk 1 above)
Comment (ssciolla) [reply src/officehours_api/backends/zoom.py:177]: Might be missing something, but this sounds serious enough that it should be documented in a separate issue. Doesn't need to be a separate PR, in my view, though.

Comment (ssciolla) [reply src/officehours_api/backends/zoom.py:177]: How would you reproduce this bug/case in QA?

Comment (jlost) [reply src/officehours_api/backends/zoom.py:177]: * Create a queue with two hosts
* Create a meeting and assign it to the second host
* Start the meeting
* Correct behavior: underlying Zoom meeting is owned by the second host. Bad behavior: Is owned by first host.

#283

Comment (ssciolla) [reply src/officehours_api/backends/zoom.py:177]: Awesome, thanks. I'll check to make sure it's working.

Comment (ssciolla) [reply src/officehours_api/backends/zoom.py:177]: I might have to wait to test this until it's in QA, unless you've pushed this to `dev`. I don't have access to two umich accounts to use locally.

File: src/officehours_api/backends/zoom.py [hunk 3]
```diff
@@ -118,6 +118,13 @@ def _get_access_token(cls, user: User) -> str:
                 },
                 headers=cls._get_client_auth_headers(),
             )
+            if resp.status_code == 401:
+                # The refresh_token was invalidated somehow.
+                # Maybe the user removed our Zoom app's auth.
+                # Force them to be prompted again.
+                zoom_meta = {}
+                user.profile.backend_metadata['zoom'] = zoom_meta
+                user.profile.save()
```
Comment (ssciolla) [inline src/officehours_api/backends/zoom.py:127]: Yeah, so this seems to work when I remove the app integration manually. It is triggered when you hit "Create Meeting". There is still an error logged in the backend; that may not be desirable, but I don't expect this to happen very often.

**Review by ssciolla — APPROVED** [review]
All these changes make sense to me. I was only able to test the changes for issue #275 locally, but I don't foresee any further issues with #283 with these changes.

File: src/officehours_api/backends/zoom.py (hunk 1 above)
Comment (jlost) [reply src/officehours_api/backends/zoom.py:177]: Pushed it to dev.

**Pull request merged**